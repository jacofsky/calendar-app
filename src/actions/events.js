import { types } from "../types/types";
import { fetchConToken } from "../helpers/fetch";
import Swal from "sweetalert2";
import { prepareEvents } from "../helpers/prepareEvents";

export const eventStartAddNew = (event) => {
    return async(dispatch, getState) => {

        const {uid, name} = getState().auth

        try {
            const resp = await fetchConToken('events', event, 'POST')
            const body = await resp.json()

            if (body.ok) {

                event.id = body.evento.id
                event.user = {
                    _id: uid,
                    name: name
                }
                dispatch(eventAddNew(body.evento))
            }
            
        } catch (error) {
            console.log('error')
            Swal.fire('Error', 'Ah surgido un problema misterioso! ðŸ˜°', 'error')
        }
    }
}

export const eventStartLoading = () => {
    return async(dispatch) =>{

        try {
            const resp = await fetchConToken('events')
            const body = await resp.json()

            if (body.ok) {
                const events = (prepareEvents(body.msg))
                dispatch(eventsLaoded(events))
            }

        } catch (error) {
            console.log('error')
            Swal.fire('Error', 'Ah surgido un problema misterioso! ðŸ˜°', 'error')
        }
    }
}

export const eventStartUpdate = (event) => {
    return async(dispatch) => {

        try {
            const resp = await fetchConToken(`events/${event.id}`, event, 'PUT')
            const body = await resp.json()

            console.log(body)

            if (body.ok) {
                dispatch(eventUpdated(event))
            } else {
                Swal.fire('Error', body.msg, 'error')
            }

        } catch (error) {
            console.log('error')
        }
    }
}

export const eventStartDelete = () => {
    return async(dispatch, getState) => {

        const  {id} = getState().calendar.activeEvent

        try {
            const resp = await fetchConToken(`events/${id}`, {}, 'DELETE')
            const body = await resp.json()

            console.log(body)

            if (body.ok) {
                dispatch(eventDeleted())
            } else {
                Swal.fire('Error', body.msg, 'error')
            }

        } catch (error) {
            console.log('error')
        }
    }
}

export const eventLogout = () => ({
    type: types.eventLogout
})

export const eventSetActive = (event) => ({
    type: types.eventSetActive,
    payload: event
})

export const eventClearActiveEvent = () => ({
    type: types.eventClearActiveEvent
})

const eventDeleted = () => ({
    type: types.eventDeleted
})

const eventUpdated = (event) => ({
    type: types.eventUpdated,
    payload: event
})

const eventsLaoded = (eventos) => ({
    type: types.eventLoaded,
    payload: eventos
})

const eventAddNew = (event) => ({
    type: types.eventAddNew,
    payload: event
})

