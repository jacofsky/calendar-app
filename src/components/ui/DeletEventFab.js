import React from 'react'
import { useDispatch } from 'react-redux'
import { eventDeleted } from '../../actions/events'

const DeletEventFab = () => {

    const dispatch = useDispatch()

    const handleClick = () => {
        dispatch(eventDeleted())
    }

  return (
    <button className='btn btn-danger fab-danger' onClick={handleClick}>
        <i className="bi bi-trash3"></i>
    </button>
  )
}

export default DeletEventFab