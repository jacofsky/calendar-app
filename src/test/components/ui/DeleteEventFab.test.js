import React from 'react'
import DeletEventFab from '../../../components/ui/DeletEventFab'
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'

import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import { eventStartDelete } from '../../../actions/events';


jest.mock(('../../../actions/events'), () => ({
    eventStartDelete: jest.fn()
}))

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initState = {}

let store = mockStore(initState)
store.dispatch = jest.fn()

const wrapper = mount(
    <Provider store={store}>
        <DeletEventFab/>
    </Provider>
)

describe('Test in <DeleteEventFab/>', () => { 
    
    test('should first', () => { 
        expect(wrapper).toMatchSnapshot()
    })

    test('should call eventStartDelete', () => { 
        wrapper.find('button').simulate('click') 
        
        expect(store.dispatch).toHaveBeenCalledWith(eventStartDelete())
    })

})