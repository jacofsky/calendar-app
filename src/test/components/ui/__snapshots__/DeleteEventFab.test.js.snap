// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <DeleteEventFab/> should first 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="bi bi-trash3"
  />
</button>
`;
